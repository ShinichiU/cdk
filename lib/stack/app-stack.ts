import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { ShortEnvironments } from '../type/env';
import { Route53 } from '../construct/route53';
import { CloudFront } from '../construct/cloudfront';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

interface Props extends cdk.StackProps {
  shortEnv: ShortEnvironments;
}
export class AppStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id, props);

    new Route53(this, 'Route53', {
      shortEnv: props.shortEnv,
    });
    new CloudFront(this, 'CloudFront', {
      shortEnv: props.shortEnv,
    });
  }
}
