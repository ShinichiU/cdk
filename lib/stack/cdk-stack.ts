import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Pipeline } from '../construct/pipeline';
import { CdkCi } from '../construct/ci';
import { GithubCredentials } from '../construct/github';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class CdkStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new Pipeline(this, 'CdkPipeline');
    const credentials = new GithubCredentials(this, 'GithubCredentials');
    const ci = new CdkCi(this, 'CdkCi');
    ci.node.addDependency(credentials);
  }
}
