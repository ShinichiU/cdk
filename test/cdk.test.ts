import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { CdkStack } from '../lib/stack/cdk-stack';
import { AppStack } from '../lib/stack/app-stack';

test(`Matches the snapshot rootStack`, () => {
  const app = new cdk.App();
  expect(
    Template.fromStack(
      new CdkStack(app, 'RootStack', {
        env: {
          account: '700359865376',
          region: 'us-east-1',
        },
      }),
    ).toJSON(),
  ).toMatchSnapshot();
});

test(`Matches the snapshot appStack`, () => {
  const app = new cdk.App();
  expect(
    Template.fromStack(
      new AppStack(app, 'AppStack', {
        env: {
          account: '992382384155',
          region: 'us-east-1',
        },
        shortEnv: 'prd',
      }),
    ).toJSON(),
  ).toMatchSnapshot();
});
