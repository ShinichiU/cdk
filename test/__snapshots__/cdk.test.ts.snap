// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot stack 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CdkPipelinecdkpipelineCodeBuildActionRole19AB58C7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineRole7C58051F",
                    "Arn",
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CdkPipelinecdkpipelineCodeBuildActionRoleDefaultPolicy1564AD05": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkPipelinecdkpipelineCodeBuildActionRoleDefaultPolicy1564AD05",
        "Roles": [
          {
            "Ref": "CdkPipelinecdkpipelineCodeBuildActionRole19AB58C7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "Description": "Pipeline step RootStack/cdk-pipeline-pipeline/UpdatePipeline/SelfMutate",
        "EncryptionKey": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
            "Arn",
          ],
        },
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:7.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelineUpdatePipelineSelfMutationRole0AE57EDA",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "{
  "version": "0.2",
  "phases": {
    "install": {
      "commands": [
        "npm install -g aws-cdk@2"
      ]
    },
    "build": {
      "commands": [
        "cdk -a . deploy RootStack --require-approval=never --verbose"
      ]
    }
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "CdkPipelinecdkpipelineUpdatePipelineSelfMutationRole0AE57EDA": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CdkPipelinecdkpipelineUpdatePipelineSelfMutationRoleDefaultPolicy1B0647DC": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:us-east-1:700359865376:log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:us-east-1:700359865376:log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:us-east-1:700359865376:report-group/",
                    {
                      "Ref": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Condition": {
                "ForAnyValue:StringEquals": {
                  "iam:ResourceTag/aws-cdk:bootstrap-role": [
                    "image-publishing",
                    "file-publishing",
                    "deploy",
                  ],
                },
              },
              "Effect": "Allow",
              "Resource": "arn:*:iam::700359865376:role/*",
            },
            {
              "Action": "cloudformation:DescribeStacks",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "s3:ListBucket",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationRoleDefaultPolicy1B0647DC",
        "Roles": [
          {
            "Ref": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationRole0AE57EDA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkPipelinecdkpipelineconnection239684AE": {
      "Properties": {
        "ConnectionName": "cdk-pipeline-connection",
        "ProviderType": "GitHub",
      },
      "Type": "AWS::CodeConnections::Connection",
    },
    "CdkPipelinecdkpipelinepipeline9E2E7AD8": {
      "DependsOn": [
        "CdkPipelinecdkpipelinepipelineRoleDefaultPolicy959CAD53",
        "CdkPipelinecdkpipelinepipelineRole7C58051F",
      ],
      "Properties": {
        "ArtifactStore": {
          "EncryptionKey": {
            "Id": {
              "Fn::GetAtt": [
                "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                "Arn",
              ],
            },
            "Type": "KMS",
          },
          "Location": {
            "Ref": "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
          },
          "Type": "S3",
        },
        "PipelineType": "V2",
        "RestartExecutionOnUpdate": false,
        "RoleArn": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelinepipelineRole7C58051F",
            "Arn",
          ],
        },
        "Stages": [
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Source",
                  "Owner": "AWS",
                  "Provider": "CodeStarSourceConnection",
                  "Version": "1",
                },
                "Configuration": {
                  "BranchName": "main",
                  "ConnectionArn": {
                    "Fn::GetAtt": [
                      "CdkPipelinecdkpipelineconnection239684AE",
                      "ConnectionArn",
                    ],
                  },
                  "DetectChanges": true,
                  "FullRepositoryId": "ShinichiU/cdk",
                },
                "Name": "ShinichiU_cdk",
                "OutputArtifacts": [
                  {
                    "Name": "ShinichiU_cdk_Source",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRole7703306B",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "EnvironmentVariables": "[{"name":"_PROJECT_CONFIG_HASH","type":"PLAINTEXT","value":"1cc7bc875bcfca93fd092681b02f6c1613a6083f3e46f216c9362cd8092dc401"}]",
                  "ProjectName": {
                    "Ref": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "ShinichiU_cdk_Source",
                  },
                ],
                "Name": "Synth",
                "OutputArtifacts": [
                  {
                    "Name": "Synth_Output",
                  },
                ],
                "RoleArn": {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelineCodeBuildActionRole19AB58C7",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          {
            "Actions": [
              {
                "ActionTypeId": {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": {
                  "EnvironmentVariables": "[{"name":"_PROJECT_CONFIG_HASH","type":"PLAINTEXT","value":"f318ce804ba3d6ddc25d26b8bb7cb4796cb0adeb948073c82267275cad331563"}]",
                  "ProjectName": {
                    "Ref": "CdkPipelinecdkpipelineUpdatePipelineSelfMutationE4B0AF47",
                  },
                },
                "InputArtifacts": [
                  {
                    "Name": "Synth_Output",
                  },
                ],
                "Name": "SelfMutate",
                "RoleArn": {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelineCodeBuildActionRole19AB58C7",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "UpdatePipeline",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "KMSMasterKeyID": {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                    "Arn",
                  ],
                },
                "SSEAlgorithm": "aws:kms",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::700359865376:root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Delete",
    },
    "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKeyAlias98C1231F": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AliasName": "alias/codepipeline-rootstackcdkpipelinecdkpipelinepipeline0ae70f67",
        "TargetKeyId": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
            "Arn",
          ],
        },
      },
      "Type": "AWS::KMS::Alias",
      "UpdateReplacePolicy": "Delete",
    },
    "CdkPipelinecdkpipelinepipelineArtifactsBucketPolicy1D03CBB1": {
      "Properties": {
        "Bucket": {
          "Ref": "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C": {
      "Properties": {
        "Artifacts": {
          "Type": "CODEPIPELINE",
        },
        "Cache": {
          "Type": "NO_CACHE",
        },
        "Description": "Pipeline step RootStack/cdk-pipeline-pipeline/Build/Synth",
        "EncryptionKey": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
            "Arn",
          ],
        },
        "Environment": {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:7.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "ServiceRole": {
          "Fn::GetAtt": [
            "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProjectRole4D1A729E",
            "Arn",
          ],
        },
        "Source": {
          "BuildSpec": "{
  "version": "0.2",
  "phases": {
    "install": {
      "commands": [
        "npm install -g aws-cdk"
      ]
    },
    "build": {
      "commands": [
        "npm ci",
        "npx prettier --check './{bin,lib,src,tests}/**/*.{ts,tsx}'",
        "npx eslint './{bin,lib,src,tests}/**/*.{ts,tsx}' --max-warnings=0",
        "npm run test",
        "npm run build",
        "npx cdk synth"
      ]
    }
  },
  "artifacts": {
    "base-directory": "cdk.out",
    "files": "**/*"
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProjectRole4D1A729E": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProjectRoleDefaultPolicy28F2D10F": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:us-east-1:700359865376:log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:us-east-1:700359865376:log-group:/aws/codebuild/",
                      {
                        "Ref": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:",
                    {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:us-east-1:700359865376:report-group/",
                    {
                      "Ref": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProject9D1F542C",
                    },
                    "-*",
                  ],
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProjectRoleDefaultPolicy28F2D10F",
        "Roles": [
          {
            "Ref": "CdkPipelinecdkpipelinepipelineBuildSynthCdkBuildProjectRole4D1A729E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkPipelinecdkpipelinepipelineRole7C58051F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CdkPipelinecdkpipelinepipelineRoleDefaultPolicy959CAD53": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRole7703306B",
                  "Arn",
                ],
              },
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelineCodeBuildActionRole19AB58C7",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkPipelinecdkpipelinepipelineRoleDefaultPolicy959CAD53",
        "Roles": [
          {
            "Ref": "CdkPipelinecdkpipelinepipelineRole7C58051F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRole7703306B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::700359865376:root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRoleDefaultPolicy4AA1D1C0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "codestar-connections:UseConnection",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelineconnection239684AE",
                  "ConnectionArn",
                ],
              },
            },
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject",
                "s3:PutObjectLegalHold",
                "s3:PutObjectRetention",
                "s3:PutObjectTagging",
                "s3:PutObjectVersionTagging",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "CdkPipelinecdkpipelinepipelineArtifactsBucketEncryptionKey2C26E8FF",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "s3:PutObjectAcl",
                "s3:PutObjectVersionAcl",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    {
                      "Fn::GetAtt": [
                        "CdkPipelinecdkpipelinepipelineArtifactsBucketDB57F132",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRoleDefaultPolicy4AA1D1C0",
        "Roles": [
          {
            "Ref": "CdkPipelinecdkpipelinepipelineSourceShinichiUcdkCodePipelineActionRole7703306B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
